mongo_connector/oplog_manager.py

```diff
@@ -260,67 +260,24 @@ def run(self):
-
-                                # Remove
-                                if operation == "d":
-                                    docman.remove(entry["o"]["_id"], ns, timestamp)
-                                    remove_inc += 1
-
-                                # Insert
-                                elif operation == "i":  # Insert
-                                    # Retrieve inserted document from
-                                    # 'o' field in oplog record
-                                    doc = entry.get("o")
-                                    # Extract timestamp and namespace
-                                    if is_gridfs_file:
-                                        db, coll = ns.split(".", 1)
-                                        gridfile = GridFSFile(
-                                            self.primary_client[db][coll], doc
-                                        )
-                                        docman.insert_file(gridfile, ns, timestamp)
-                                    else:
-                                        docman.upsert(doc, ns, timestamp)
-                                    upsert_inc += 1
-
-                                # Update
-                                elif operation == "u":
-                                    docman.update(
-                                        entry["o2"]["_id"], entry["o"], ns, timestamp
-                                    )
-                                    update_inc += 1
-
-                                # Command
-                                elif operation == "c":
-                                    # use unmapped namespace
-                                    doc = entry.get("o")
-                                    docman.handle_command(doc, entry["ns"], timestamp)
-
-                            except errors.OperationFailed:
-                                LOG.exception(
-                                    "Unable to process oplog document %r" % entry
-                                )
-                            except errors.ConnectionFailed:
-                                LOG.exception(
-                                    "Connection failed while processing oplog "
-                                    "document %r" % entry
-                                )
-
-                        if (remove_inc + upsert_inc + update_inc) % 1000 == 0:
-                            LOG.debug(
-                                "OplogThread: Documents removed: %d, "
-                                "inserted: %d, updated: %d so far"
-                                % (remove_inc, upsert_inc, update_inc)
-                            )
+
+                        if self.is_transaction_entry(entry):
+                            operations = entry['o'].get('applyOps', [])
+                            for _entry in operations:
+                                remove_inc, upsert_inc, update_inc = self.process_operation(_entry,
```

Comment:
It'd be better to keep the counters outside the process function
so th eprocess function will return how many operations they mad, and inside the WHILE loop we increment the counter and log them
This is to make the process function's scope limited to processing the operations and returning the count of operations made. Also, to avoid modifying the inputs to the process function where we do sth like `remove_inc += 1`

Reply:
Thanks @AhmedFat7y for your review on the PR.
I've update the PR and will merge it to master, Feel free to re-open any further request if you want as issue and I will start work on it immediately.

Thanks again